pkg/adscert/authenticated_connections_signer_api.go

```diff
@@ -12,6 +12,7 @@ type AuthenticatedConnectionsSigner interface {
 	SignAuthenticatedConnection(params AuthenticatedConnectionSignatureParams) (AuthenticatedConnectionSignature, error)
 
 	VerifyAuthenticatedConnection(params AuthenticatedConnectionSignatureParams) (AuthenticatedConnectionVerification, error)
+	VerifyAuthenticatedConnectionWithPackage(params adscertcrypto.AuthenticatedConnectionVerificationPackage) (AuthenticatedConnectionVerification, error)
```

Comment:
I was a bit back and forth on creating a new function vs extending the existing one but looking at it again, since they're largely functionally equivalent, I think it would make sense to extend the original. I will update.

Reply:
Yes, if you don't mind incorporating this function into the VerifyAuthenticatedConnection function as separate hash parameters in the AuthenticatedConnectionSignatureParams message, that would be preferred from my perspective.  The "params" message here acts more like an RPC message argument than a simple function call due to the desire to let us change the implementation without creating compile time errors in callers.

It should be fine to defer this to another pull request if you prefer, but I was thinking it could simply be, in the assembleRequestInfo() function:

if bodyHash populated
  set the body hash directly
else
  calculate the body hash

etc

Reply:
Something we should discuss outside of this pull request is whether this API is indeed suitable for batch processing or if we want to restructure it so that it can handle batches of signature verifications in one operation.  When I originally wrote this, I didn't have the batch use case in mind, but it might make sense.